changelog.d/rules-6457.changed

```diff
@@ -0,0 +1,14 @@
+taint-mode: Improve inference of best matches for exact-sources, exact-sanitizers,
+and sinks. Now we also avoid FPs in cases such as:
+
+    dangerouslySetInnerHTML = {
+      // ok:
+      {__html: props ? DOMPurify.sanitize(props.text) : ''} // no more FPs!
```

Comment:
I'm unsure how the more accurate match in the sink, as demonstrated in the PR description, results in the removal of false positives in this example. Wouldn't a more accurate match in the sink improve the detection of missed findings (false negatives)? Perhaps I'm overlooking something straightforward 😅 Could you please provide additional details or clarification regarding the example in the changelog?